app/assets/javascripts/admin/dataset/dataset_add_foreign_view.js

```diff
@@ -0,0 +1,137 @@
+// @flow
+import _ from "lodash";
+import React from "react";
+import { Form, Input, Button, Card, Spin } from "antd";
+import { addForeignDataSet } from "admin/admin_rest_api";
+import Messages from "messages";
+import Toast from "libs/toast";
+import { withRouter } from "react-router-dom";
+
+const FormItem = Form.Item;
+const TextArea = Input.TextArea;
+
+type Props = {
+  form: Object,
+};
+
+type State = {
+  isUploading: boolean,
+};
+
+export type ForeignDataSetSpecification = {
+  +dataStoreName: string,
+  +url: string,
+  +dataSetName: string,
+};
+
+class DatasetAddForeignView extends React.PureComponent<Props, State> {
+  state = {
+    isUploading: false,
+  };
+
+  isValidDataSet(specification: ForeignDataSetSpecification): boolean {
+    if (
+      !_.isString(specification.dataStoreName) ||
+      !_.isString(specification.url) ||
+      !_.isString(specification.dataSetName) ||
+      !/^[A-Za-z0-9_-]*$/.test(specification.dataSetName) ||
+      !/^https?:\/\/[a-z0-9.]+.*$/.test(specification.url)
+    ) {
+      return false;
+    }
+    return true;
+  }
+
+  splitToWords(string: string): Array<string> {
+    return string
+      .split(",")
+      .map(word => word.trim())
+      .filter(word => word !== "");
+  }
+
+  parseLine(specification: string): ForeignDataSetSpecification {
+    const words = this.splitToWords(specification);
+    const dataStoreName = words[0];
+    const url = words[1];
+    const dataSetName = words[2];
+
+    return {
+      dataStoreName,
+      url,
+      dataSetName,
+    };
+  }
+
+  handleSubmit = async e => {
+    e.preventDefault();
+
+    const formValues = this.props.form.getFieldsValue();
+    const specification = this.parseLine(formValues.foreignDatasetText);
+
+    if (this.isValidDataSet(specification)) {
+      await addForeignDataSet(
+        specification.dataStoreName,
+        specification.url,
+        specification.dataSetName,
+      );
+      window.location.replace("/dashboard");
```

Comment:
I think it would be better to handle the successful addition of a dataset in the same way as the successful upload of a dataset, using a callback prop. You could add an `onAdded` prop to this component, which you call here. Then, in the `dataset_add_view` you set the prop in the same way as for the `DatasetUploadView`, like this:
```
<DatasetAddForeignView onAdded={() => history.push("/dashboard")} />
```
Two advantages, history.push doesn't cause a page reload as its using react routers single-page routing, and the `DatasetAddForeignView` can be easily reused somewhere else, for example, as part of the onboarding flow (where you don't want to redirect to the dashboard).